src/treetensornetworks/solvers/contract_mpo_mps.jl

```diff
@@ -30,16 +30,20 @@ function ITensors.contract(
   init_mps = deepcopy(init_mps)
   init_mps = sim(linkinds, init_mps)
   Ai = siteinds(A)
-  ti = Vector{Index}(undef, n)
+  init_mpsi = siteinds(init_mps)
   for j in 1:n
+    ti = nothing
     for i in Ai[j]
       if !hasind(psi0[j], i)
-        ti[j] = i
+        ti = i
         break
       end
     end
+    if ti !== nothing
+      ci = commoninds(init_mpsi[j], A[j])[1]
+      replaceind!(init_mps[j], ci=>ti)
+    end
   end
```

Comment:
Could you explain what this is doing?

Reply:
This ensures that `phi0` and `init_mps` have the same site indices. This originates from the old code, but I feel this is hard to understand. Why not throwing an error if the two MPSs do not have the site indices?

Reply:
In general, I prefer to write the code where it is as flexible as possible about the indices, so it should be ok if there are missing site indices (for example I have hit examples like that in `ITensorQTT.jl`).

Reply:
Along those lines, in general I would avoid doing index replacements/forcing indices to match, if that is what this code is doing.

Reply:
Oh, there was a typo. This code ensures that `A phi0` and `init_mps` have the same site indices.
Which do you prefer? I guess you prefer No. 2.

1. Do not check if `A phi0` and `init_mps` do not have the same site indices.
2. Call `error()` if `A phi0` and `init_mps` do not have the same site indices.
3. Replace the site indices of `init_mps` if `A phi0` and `init_mps` do not have the same site indices.

Reply:
I would vote for just not checking or fixing anything at first to allow the most flexibility, and if issues come up from that we can add checks for particular issues.

Reply:
For context, I think in general in the past we have over-specialized code too much in functions like this, and the goal of ITensorNetworks.jl is to be as agnostic as possible about particular tensor network structures.

Reply:
Along those lines, as of https://github.com/mtfishman/ITensorNetworks.jl/pull/44 the underlying generic tdvp function now works for tree tensor networks, so with some minor modifications it should be easy to make this function work for that case as well!